src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,54 @@
+import { auth } from "../../config/firebase.js";
+import {
+  createUserWithEmailAndPassword,
+  signOut,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { useState } from "react";
+
+export const useAuth = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const register = async () => {
+    try {
+      await createUserWithEmailAndPassword(auth, email, password);
+    } catch (err) {
+      alert(err);
+    }
+  };
+
+  const login = async () => {
+    try {
+      await signInWithEmailAndPassword(auth, email, password);
+    } catch (err) {
+      alert(err);
+    }
+  };
+
+  const logout = async () => {
+    try {
+      await signOut(auth);
+    } catch (err) {
+      alert(err);
+    }
+  };
+
+  return {
+    setEmail,
+    setPassword,
+    register,
+    logout,
+    login,
+  };
+};
```

Comment:
Please log potential errors by console.log, users don't have to see them everytime. What about checking if user is logged in?